src/metabase/lib/field.cljc

```diff
@@ -338,8 +338,9 @@
   (cond->> options
     (some #(= (:unit %) unit) options)
     (mapv (fn [option]
-            (cond-> (dissoc option option-key)
-              (= (:unit option) unit) (assoc option-key true))))))
+            (cond-> option
+              (contains? option option-key) (dissoc option option-key)
+              (= (:unit option) unit)       (assoc option-key true))))))
```

Comment:
Why is this change needed?

Reply:
It's a leftover and isn't vital in the final version. I was trying to identity-cache various permutations, and found that while the returned buckets were practically identical, the original logic to always `dissoc` meant the bucket/bins returned on columns were all not `identical?` even though `=`. That seemed silly, when they're coming from a static `def` and then maybe one of them gets a `:selected` or `:default` set on it.

Reply:
Interesting, I've checked in a node REPL, and there
```
cljs.user=> (let [m {:x 1}] (identical? m (dissoc m :y)))
true
```